src/app/mocks.spec.ts

```diff
@@ -0,0 +1,15 @@
+import { mockProvider } from '@netbasal/spectator';
+import { WidgetService } from './widget.service';
+
+describe('mockProvider', () => {
+  it('should not modify the object passed in 2nd argument when running the mock factory', () => {
+    const customPropertiesAndMethods: Partial<Record<keyof WidgetService, any>> = {
+      testingProperty: 'overriden'
+    };
+    const { useFactory: factory } = mockProvider(WidgetService, customPropertiesAndMethods);
+    factory();
+    expect(customPropertiesAndMethods).toEqual({
+      testingProperty: 'overriden'
+    });
+  });
+});
```

Comment:
This is really a unit test for `mockProvider` function. Imho should be located somewhere close to `mock.ts` but there is no place for unit tests. I placed it here but it should be probably moved in the future.